nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2024 Angsa Robotics
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <memory>
+#include <limits>
+
+#include "nav2_behavior_tree/plugins/action/remove_in_collision_goals_action.hpp"
+#include "nav2_behavior_tree/bt_utils.hpp"
+#include "tf2/utils.h"
+#include "tf2_geometry_msgs/tf2_geometry_msgs.hpp"
+
+namespace nav2_behavior_tree
+{
+
+RemoveInCollisionGoals::RemoveInCollisionGoals(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::ActionNodeBase(name, conf),
+  initialized_(false),
+  which_costmap_("both"),
+  cost_threshold_(253)
+{}
+
+void RemoveInCollisionGoals::initialize()
+{
+  node_ = config().blackboard->get<rclcpp::Node::SharedPtr>("node");
+  get_global_cost_client_ = node_->create_client<nav2_msgs::srv::GetCost>(
+    "global_costmap/get_cost_global_costmap");
+  get_local_cost_client_ = node_->create_client<nav2_msgs::srv::GetCost>(
+    "local_costmap/get_cost_local_costmap");
+  server_timeout_ = config().blackboard->template get<std::chrono::milliseconds>("server_timeout");
+}
+
+inline BT::NodeStatus RemoveInCollisionGoals::tick()
+{
+  setStatus(BT::NodeStatus::RUNNING);
+
+  if (!initialized_) {
+    initialize();
+  }
+
+  getInput("costmap", which_costmap_);
```

Comment:
Do you think you'd change costmaps to use mid-BT execution for a given instance of this node? If not, we can getInput that in the `initialize()` method and only create the global or local costmap service instead of both

Reply:
I see there's a 'both' option - do you think you'd want / need that for a single given instance of this BT node?

Reply:
> I see there's a 'both' option - do you think you'd want / need that for a single given instance of this BT node?

I'm changing it so that it only checks on one service and in the BT we can chain them together if we want to check on both costmaps